truffle/src/main/java/org/jruby/truffle/language/WarnNode.java

```diff
@@ -0,0 +1,27 @@
+package org.jruby.truffle.language;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.object.DynamicObject;
+import org.jcodings.specific.UTF8Encoding;
+import org.jruby.truffle.language.dispatch.CallDispatchHeadNode;
+
+public class WarnNode extends RubyBaseNode {
+
+    @Child CallDispatchHeadNode warnMethod = CallDispatchHeadNode.createMethodCall();
+
+    public Object execute(VirtualFrame frame, Object... arguments) {
```

Comment:
Since arguments seem to be always String for now, I would type them as such.
`toString` is usually a bad practice.